Orange/preprocess/discretize.py

```diff
@@ -278,40 +297,174 @@ def unique_bins():
         nbins = np.round((mx_ - mn_) / width)
         if min_bins <= nbins <= max_bins \
                 and (not bins or bins[-1].nbins != nbins):
-            bins.append(BinDefinition(mn_, nbins, width))
+            bin_def = BinDefinition(mn_ + width * np.arange(nbins + 1),
+                                    label_fmt, width)
+            bins.append(bin_def)
+    return bins
 
-    if add_unique is not None and len(unique) <= add_unique:
-        if bins and bins[-1].nbins == len(unique):
-            del bins[-1]
-        bins.append(unique_bins())
-        if len(unique) < min_unique:
-            del bins[:-1]
 
-    if not return_defs:
-        bins = [get_bins(bin) for bin in bins]
+def time_binnings(data, *, min_bins=2, max_bins=50, min_unique=5, add_unique=0):
```

Comment:
The output of `time_binnings` and `decimal_binnings` are not the same. The latter uses `BinDefinition`, which has thresholds and labels properties, while the former doesn't.

Reply:
This can't be. OWDistributions calls either `time_binnings` or `decimal_binnings`, stores the result and then doesn't care where the bins came from.

Reply:
But OWSOM doesn't.

Reply:
This was OWSOM's problem. Fixed.